plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/applicationexplorer/CreateComponentHandler.java

```diff
@@ -44,31 +48,46 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
 		ISelection selection = HandlerUtil.getCurrentSelection(event);
 		ApplicationElement application = null;
 		ProjectElement project = UIUtils.getFirstElement(selection, ProjectElement.class);
+		DevfileRegistryComponentTypeElement componentType = null;
 		if (project == null) {
 			application = UIUtils.getFirstElement(selection, ApplicationElement.class);
 			if (application == null) {
-				return OpenShiftUIActivator.statusFactory().cancelStatus("No project or application selected"); //$NON-NLS-1$
+			  componentType = UIUtils.getFirstElement(selection, DevfileRegistryComponentTypeElement.class);
+			  if (componentType == null) {
+			    DevfileRegistryComponentTypeStarterElement starter = UIUtils.getFirstElement(selection, DevfileRegistryComponentTypeStarterElement.class);
+			    if (starter == null) {
+		        return OpenShiftUIActivator.statusFactory().cancelStatus("No project or application selected"); //$NON-NLS-1$
+			    } else {
+			      componentType = starter.getParent();
+			    }
+			  }
+			} else {
+	      project = application.getParent();
 			}
-			project = application.getParent();
 		}
     final Shell parent = HandlerUtil.getActiveShell(event);
 		try {
-			openDialog(application, project, parent);
+			openDialog(componentType, application, project, parent);
```

Comment:
I think there is a missing arg, because when I want to create a component from starter, the dialog does not show it on the project starter droplist, only the componentType is selected.